src/sentry/models/organization.py

```diff
@@ -188,12 +191,20 @@ def __str__(self):
 
     def save(self, *args, **kwargs):
         if not self.id:
-            self.id = snowflake_id_generation()
+            self.id = snowflake_id_generation(self.snowflake_redis_key)
         if not self.slug:
             lock = locks.get("slug:organization", duration=5)
             with TimedRetryPolicy(10)(lock.acquire):
                 slugify_instance(self, self.name, reserved=RESERVED_ORGANIZATION_SLUGS)
-        super().save(*args, **kwargs)
+        try:
+            with transaction.atomic():
+                super().save(*args, **kwargs)
+        except IntegrityError:
+            self.snowflake_retry_counter += 1
+            if self.snowflake_retry_counter == 5:
+                raise Exception("Max allowed ID retry reached. Please try again in a second")
```

Comment:
@maxiuyuan This only retries at most 4 times since you're incrementing the `self.snowflake_retry_counter` prior to the if guard.

Some proposals:
- Move the `self.snowflake_retry_counter` increment to be after the if guard.
- Create `MAX_REDIS_SNOW_FLAKE_RETY_COUNTER = 5` constant so that we dont need to hardcode `5` everywhere. 

```suggestion
            if self.snowflake_retry_counter > MAX_REDIS_SNOW_FLAKE_RETY_COUNTER:
                raise Exception("Max allowed ID retry reached. Please try again in a second")
            self.snowflake_retry_counter += 1
```